File: io/src/main/java/com/xuebinduan/io/MainActivity.java
package com.xuebinduan.io;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Environment;
import android.os.FileUtils;
import android.util.Log;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;

public class MainActivity extends AppCompatActivity {
    public String SOURCE_FILE_PATH ;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SOURCE_FILE_PATH = getExternalFilesDir("").getAbsolutePath()+File.separator+"text.txt";
        Log.e("TAG","查看路径："+new File(SOURCE_FILE_PATH).getAbsolutePath());



    }
}

File: custom_view/drawable/src/main/java/com/xuebinduan/drawable/MainActivity.kt
package com.xuebinduan.drawable

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        startActivity(Intent(this,ShowTextCursorDrawableActivity::class.java))

        val view = View(this)
        Log.e("TAG","view toString:"+view.toString())


    }
}